File: frontend/src/services/receptionistService.js
import apiClient from './apiClient';

const receptionistService = {
  async getAll(params = {}) {
    const { data } = await apiClient.get('/receptionists', { params });
    return data.data;
  },
  async getById(id) {
    const { data } = await apiClient.get(`/receptionists/${id}`);
    return data.data;
  },
  async create(payload) {
    const { data } = await apiClient.post('/receptionists', payload);
    return data.data;
  },
  async update(id, payload) {
    const { data } = await apiClient.put(`/receptionists/${id}`, payload);
    return data.data;
  },
  async remove(id) {
    await apiClient.delete(`/receptionists/${id}`);
  },
};

export default receptionistService;

File: backend/src/main/java/com/hospital/hms/service/ReceptionistService.java
package com.hospital.hms.service;

import com.hospital.hms.dto.request.ReceptionistRequest;
import com.hospital.hms.dto.response.ReceptionistResponse;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

public interface ReceptionistService {

    ReceptionistResponse create(ReceptionistRequest request);

    ReceptionistResponse update(Long id, ReceptionistRequest request);

    ReceptionistResponse getById(Long id);

    Page<ReceptionistResponse> getAll(Pageable pageable);

    void delete(Long id);
}
